### PR TITLE
Adds stealthy Bikehorn sword to traitor Clown

### PR DESCRIPTION
**Pull request by Wilchenx**
A clown can now buy an esword disguised as a bikehorn through his uplink for 10 tc.

It actives through alt-clicking.

I know this is a straight buff to normal eswords because of the hiding factor, but fuck that only clowns have it.

🆑
rscadd: adds a new clown traitor item: A bikehorn esword. Activate it by alt-clicking.
/🆑

https://user-images.githubusercontent.com/7501474/29424708-0ea53cfe-8381-11e7-84c2-1d162d528868.png

original code by Time-Green

### CODE REVIEW

File: code/game/objects/items/melee/energy.dm [hunk 1]
```diff
@@ -228,3 +228,62 @@
...
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+	/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: Initialize and . = ..()

File: code/game/objects/items/melee/energy.dm [hunk 2]
```diff
@@ -228,3 +228,62 @@
...
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: use update_icon

File: code/game/objects/items/melee/energy.dm [hunk 3]
```diff
@@ -228,3 +228,62 @@
 	desc = "An extremely sharp blade made out of hard light. Packs quite a punch."
 	icon_state = "lightblade"
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: don't create the horn here

File: code/game/objects/items/melee/energy.dm [hunk 4]
```diff
@@ -228,3 +228,62 @@
...
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+	/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
+	horn.forceMove(src)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: create the horn here with src as the loc and remove this line

File: code/game/objects/items/melee/energy.dm [hunk 5]
```diff
@@ -228,3 +228,62 @@
...
+	/obj/item/weapon/melee/transforming/energy/sword/bikehorn/examine(mob/user)
+	if(!active)
+		horn.examine(user)
+	else
+		..()
+	
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/attack_self(mob/living/carbon/L)
+	if(canSword)
+		activate(L)
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: playsound(src, ...

File: code/game/objects/items/melee/energy.dm [hunk 6]
```diff
@@ -228,3 +228,62 @@
...
+		icon = horn.icon
+		icon_state = horn.icon_state
+ 
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
+		canSword = !canSword
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
+		else
+ 			L << "<span class='notice'>You squeeze the [name]. It functions like a normal bikehorn again.</span>"
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: to_chat and [src]

File: code/game/objects/items/melee/energy.dm [hunk 7]
```diff
@@ -228,3 +228,62 @@
...
+	else
+		item_state = horn.item_state
+		icon = horn.icon
+		icon_state = horn.icon_state
+ 
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
+		canSword = !canSword
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: to_chat and [src]

File: code/game/objects/items/melee/energy.dm [hunk 8]
```diff
@@ -228,3 +228,62 @@
...
+		activate(L)
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
+	else
+		horn.attack_self(L)
+
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(!active)
+		horn.attack(M,user)
+	else
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: playsound(src, ...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/melee/energy.dm [hunk 9]
```diff
@@ -228,3 +228,62 @@
...
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: Initialize and . = ..()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/melee/energy.dm [hunk 10]
```diff
@@ -228,3 +228,62 @@
...
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
+		..()
+
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/Crossed(mob/living/L)
+	if(!active)
+		horn.Crossed(L)
+	else
+		..()
+ 
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/setIcon()
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: use update_icon

File: code/game/objects/items/melee/energy.dm [hunk 11]
```diff
@@ -228,3 +228,62 @@
...
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
```
Comment (Cyberboss) [inline code/game/objects/items/melee/energy.dm]: use update_icon

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by The-clown-known-as-Pepe** [comment]
still needs work but i can't do it now

**Comment by InsaneHyena** [comment]
>Downvoted by people with no sense of humour
Does it honk when you hit enemies instead of making esword noises? Please say yes.

**Comment by DoctoreMad** [comment]
Its not that I don't have a sense of humor I just think banana peel triple-e-sword is a better meme.

**Comment by Kraseo** [comment]
>a stealth bikehorn that costs the same as a regular esword for the le epic grief job

uhhhhhhhhhhhhh

**Comment by The-clown-known-as-Pepe** [comment]
@Kraseo changes to 10
@InsaneHyena yes it honks

**Comment by Super3222** [comment]
@The-clown-known-as-Pepe drag and drop images into the chat box / pr description to upload them.
also, the PR he's porting is: https://github.com/yogstation13/yogstation/pull/2483

**Comment by MoreRobustThanYou** [comment]
Port the invisible spray / clowncart / superglue from /vg/ as clown items.

**Comment by Supermichael777** [comment]
your avatar image violates our standards

- National Socialist Party of Germany content, National Socialist Party of Germany related content, or National Socialist Party of Germany references

**Comment by Kraseo** [comment]
Thanks, Superminimod777!

File: code/game/objects/items/melee/energy.dm [hunk 12]
```diff
@@ -228,3 +228,65 @@
...
+		horn.attack(M,user)
+	else
+		playsound(src, 'sound/items/bikehorn.ogg', 80, 1)
+		..()
+
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/Crossed(mob/living/L)
+	if(!active)
+		horn.Crossed(L)
+	else
+		..()
```
Comment (Qustinnus) [inline code/game/objects/items/melee/energy.dm]: .=..()

File: code/game/objects/items/melee/energy.dm [hunk 13]
```diff
@@ -228,3 +228,65 @@
 	desc = "An extremely sharp blade made out of hard light. Packs quite a punch."
 	icon_state = "lightblade"
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
```
Comment (Qustinnus) [inline code/game/objects/items/melee/energy.dm]: change all /obj/item/weapon/melee/transforming/energy/sword/bikehorn to /obj/item/melee/transforming/energy/sword/bikehorn

File: code/game/objects/items/melee/energy.dm [hunk 14]
```diff
@@ -228,3 +228,65 @@
 	desc = "An extremely sharp blade made out of hard light. Packs quite a punch."
 	icon_state = "lightblade"
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
```
Comment (Qustinnus) [inline code/game/objects/items/melee/energy.dm]: 🤔

File: code/modules/uplink/uplink_item.dm [hunk 15]
```diff
@@ -1248,6 +1248,13 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 	exclude_modes = list(/datum/game_mode/nuclear)
 	surplus = 0
 
+/datum/uplink_item/role_restricted/bikehornsword
+	name = "Energy Sword Bikehorn"
+	desc = "A bikehorn with a built-in energy sword. Activate through Alt-Click."
```
Comment (Qustinnus) [inline code/modules/uplink/uplink_item.dm]: boring description

**Comment by Chronitonity** [comment]
atleast make it semi interesting instead of adding just another esword but a bike horn, thats just lazy

**Comment by The-clown-known-as-Pepe** [comment]
it honks when ignited and on hit, its super stealthy

File: code/game/objects/items/melee/energy.dm [hunk 16]
```diff
@@ -228,3 +228,65 @@
...
+		item_state = "swordpink"
+		icon = 'icons/obj/weapons.dmi'
+		icon_state = "ehonk"
+	else
+		item_state = horn.item_state
+		icon = horn.icon
+		icon_state = horn.icon_state
+ 
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
```
Comment (kevinz000) [inline code/game/objects/items/melee/energy.dm]: i really don't like the idea of something as dangerous as an esword being able to be easily hidden without any way to detect it short of carrying research scanners.
it'll just encourage security to stuff everything in the deconner.

Comment (The-clown-known-as-Pepe) [inline code/game/objects/items/melee/energy.dm]: we don't see that done with pie cannon do we?

File: code/modules/uplink/uplink_item.dm (hunk 15 above)
Comment (Time-Green) [inline code/modules/uplink/uplink_item.dm]: No it's not, it's a good description

File: code/game/objects/items/melee/energy.dm (hunk 16 above)
Comment (WJohn) [inline code/game/objects/items/melee/energy.dm]: the existence of bad items doesn't justify adding more bad items

**Comment by Ratvar** [comment]
Make it honk 200% louder in active state, remove R&D check, allow only clowns to unflap blade. Everyone knows when it hits, everyone wants valids, nobody can check which clown did it. Change item to glued sharpended toy blade you attach to bike horn to create honksword?  Just so shitcurity cannot "Two Bikehorns in pocket policy here, prepare to die, you have 1 horn too much.".
Or add action button to convert any bikehorn/duck into sword instead of weapon anyone will check and take? Trade new sword from fellow cluwne in exchange of emag/bullet in the head.

**Comment by DoctoreMad** [comment]
In english please

File: code/game/objects/items/melee/energy.dm [hunk 17]
```diff
@@ -228,3 +228,65 @@
 	desc = "An extremely sharp blade made out of hard light. Packs quite a punch."
 	icon_state = "lightblade"
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/weapon/bikehorn = new()
```
Comment (ChangelingRain) [inline code/game/objects/items/melee/energy.dm]: don't do this, just make the horn in update_icon

**Comment by Cheridan** [comment]
uninspired
not funny

**Comment by jughu** [comment]
Sprites!

**Comment by Shadowlight213** [comment]
I'm gonna use this as justification to steal the clown's bikehorn every round as sec.

**Comment by KingPhilipIII** [comment]
@Shadowlight213  As though people don't already slip the clown to steal all his stuff already.

**Comment by Shadowlight213** [comment]
I'll have more justification to abuse the clown

**Comment by KingPhilipIII** [comment]
Fair enough.

**Comment by YakumoChen** [comment]
port clown cart and bananarangs, not this

**Comment by Cyberboss** [comment]
Yeah, not gonna do this

**Pull request closed without merging**